rust/arrow/src/json/writer.rs

```diff
@@ -660,6 +765,138 @@ mod tests {
+
+        let schema = Schema::new(vec![
+            Field::new("time32sec", arr_time32sec.data_type().clone(), false),
+            Field::new("time32msec", arr_time32msec.data_type().clone(), false),
+            Field::new("time64usec", arr_time64usec.data_type().clone(), false),
+            Field::new("time64nsec", arr_time64nsec.data_type().clone(), false),
+            Field::new("name", arr_names.data_type().clone(), false),
+        ]);
+        let schema = Arc::new(schema);
+
+        let batch = RecordBatch::try_new(
+            schema,
+            vec![
+                Arc::new(arr_time32sec),
+                Arc::new(arr_time32msec),
+                Arc::new(arr_time64usec),
+                Arc::new(arr_time64nsec),
+                Arc::new(arr_names),
+            ],
+        )
+        .unwrap();
+
+        let mut buf = Vec::new();
+        {
+            let mut writer = LineDelimitedWriter::new(&mut buf);
+            writer.write_batches(&[batch]).unwrap();
+        }
+
+        assert_eq!(
+            String::from_utf8(buf).unwrap(),
+            r#"{"time32sec":"00:02:00","time32msec":"00:00:00.120","time64usec":"00:00:00.000120","time64nsec":"00:00:00.000000120","name":"a"}
+{"name":"b"}
+"#
+        );
+    }
+
+    #[test]
+    fn write_durations() {
+        let arr_durationsec = DurationSecondArray::from(vec![Some(120), None]);
+        let arr_durationmsec = DurationMillisecondArray::from(vec![Some(120), None]);
+        let arr_durationusec = DurationMicrosecondArray::from(vec![Some(120), None]);
+        let arr_durationnsec = DurationNanosecondArray::from(vec![Some(120), None]);
+        let arr_names = StringArray::from(vec![Some("a"), Some("b")]);
+
+        let schema = Schema::new(vec![
+            Field::new("duration_sec", arr_durationsec.data_type().clone(), false),
+            Field::new("duration_msec", arr_durationmsec.data_type().clone(), false),
+            Field::new("duration_usec", arr_durationusec.data_type().clone(), false),
+            Field::new("duration_nsec", arr_durationnsec.data_type().clone(), false),
+            Field::new("name", arr_names.data_type().clone(), false),
+        ]);
+        let schema = Arc::new(schema);
+
+        let batch = RecordBatch::try_new(
+            schema,
+            vec![
+                Arc::new(arr_durationsec),
+                Arc::new(arr_durationmsec),
+                Arc::new(arr_durationusec),
+                Arc::new(arr_durationnsec),
+                Arc::new(arr_names),
+            ],
+        )
+        .unwrap();
+
+        let mut buf = Vec::new();
+        {
+            let mut writer = LineDelimitedWriter::new(&mut buf);
+            writer.write_batches(&[batch]).unwrap();
+        }
+
+        assert_eq!(
+            String::from_utf8(buf).unwrap(),
+            r#"{"duration_sec":"PT120S","duration_msec":"PT0.120S","duration_usec":"PT0.000120S","duration_nsec":"PT0.000000120S","name":"a"}
```

Comment:
I am not familiar with the `"PT120S"` string representation of a duration (perhaps this is some Chrono thing)?

I would have expected durations to have the same format as Time (probably) so instead of `PT120S` something more like `00:02:00` and instead of `PT0.120S` it would be `00:00:00.120`

What do you think?

Reply:
Yeah, this is a Chrono default. I don't have a preference on how duration should be formatted. the time format you mentioned is what I expected initially as well. Let me do some research on this to see if there is any particular reason why chrono went with the `PT` notation.

Reply:
I don't have experience with working with durations, but I'm also now curious. I'll do some research too

Reply:
Turns out this is a part of https://en.wikipedia.org/wiki/ISO_8601#Durations

Reply:
ISO standard is good enough for me 👍